### PR TITLE
Do not use `std::io::IoSlice` in libxcb's FFI

### PR DESCRIPTION
**Pull request by eduardosm**
On Windows, xcb defines its own `iovec` struct in `xcb_windefs.h`, which does not match the `WSABUF` struct used in `IoSlice`.

### CODE REVIEW

File: src/xcb_ffi/raw_ffi.rs [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-#[cfg(not(test))]
+#[cfg(not(all(test, unix)))]
 use libc::c_void;
+#[cfg(unix)]
+pub(crate) use libc::iovec;
 use libc::{c_char, c_int, c_uint};
 
+// As defined in xcb_windefs.h
+#[cfg(not(unix))]
+#[derive(Copy, Clone, Debug)]
```
Comment (psychon) [inline src/xcb_ffi/raw_ffi.rs]: Why `Copy` and `Clone`? `std::io::IoSlice` only has `Debug` and `Deref`.

**Comment by psychon** [comment]
How did you find this? Did this actually break something, or did you just come across this "accidentally"?

File: src/xcb_ffi/raw_ffi.rs (hunk 1 above)
Comment (eduardosm) [reply src/xcb_ffi/raw_ffi.rs]: But `iovec` from libc [has `Copy` and `Clone`](https://docs.rs/libc/0.2.67/libc/struct.iovec.html) (but not `Debug`).

There is a [PR](https://github.com/rust-lang/rust/issues/69395) to implemente `Copy` and `Clone` for `std::io::IoSlice`.

Anyways we are not using them and the definition is private, so we could remove them.

**Comment by eduardosm** [comment]
> How did you find this? Did this actually break something, or did you just come across this "accidentally"?

Just accidentally, I didn't have the chance to test x11rb on Windows yet.

**Pull request merged**